### DISCUSSION TITLE
How to setup the polkscan on my local?

### DISCUSSION THREAD

**Discussion by SWS-5007** [polkascan/explorer#21] [Q&A]
Hi, I am going to clone this Polkascan and Run this on my local with my database and substrate api.
How to do this?
**Please let me know step by step from scratch.**
**I am getting no senses from cloning this repo on my local.**
**And how to change the config environments with my custom infos including new database and substrate api ?**
**Can I run this polkascan on my Win_10 local system?**

Best Regards.

**Comment by arjanz** [comment]
> Can I run this polkascan on my Win_10 local system?

Is there a specific reason you don't want to use Docker to run the application? It will make things much easier and agnostic of OS.

But you can run everything on your local machine of course, you will need a Python 3.7+ (virtualenv) for the harvester and explorer and Node to run the UI. 

Then basically follow the steps mentioned at https://github.com/polkascan/explorer#local

Also be sure to add the root directories of `harvester` and `explorer-api`  to your Python path 

> And how to change the config environments with my custom infos including new database and substrate api ?

* For the explorer copy the [.env.template](https://github.com/polkascan/explorer-api/blob/5d75c6c49d6a4102b32a2c53d2195fb132a395a7/.env.template) to `.env` and modified the settings accordingly

* For the harvester modify the settings in [settings.py](https://github.com/polkascan/harvester/blob/f984b87b67ca5a1f4792cdb227b163f31f5e4d68/app/settings.py)

**Reply by SWS-5007** [reply]
Hi, Thanks so much for your reply.

I installed a Python 3,10.8 already and Node 16.13.1 on my local.
Is this okay?

And how to add the root directories of `havester `and `explorer-api` to my Python path?
Please let me know more clearly.
Best Regards.

**Reply by arjanz** [reply]
> 
> I installed a Python 3,10.8 already and Node 16.13.1 on my local. Is this okay?

Should be good to go with these versions

> And how to add the root directories of `havester `and `explorer-api` to my Python path? Please let me know more clearly. Best Regards.

I found this on Stackoverflow that might help: https://stackoverflow.com/questions/3701646/how-to-add-to-the-pythonpath-in-windows-so-it-finds-my-modules-packages

**Comment by crazydevlegend** [comment]
Were you able to successfully run Polkascan locally, @SWS-5007 ?
